api/apps/api/src/modules/scenarios/scenarios.controller.ts

```diff
@@ -170,4 +173,22 @@ export class ScenariosController {
       await this.service.getFeatures(id),
     );
   }
+
+  @Header('Content-Type', 'text/csv')
+  @ApiOkResponse({
+    schema: {
+      type: 'string',
+    },
+  })
+  @ApiOperation({
+    description: `Uploaded cost surface data`,
+  })
+  @Get(`:id/marxan/dat/pu.dat`)
+  async getScenarioCostSurface(
+    @Param('id', ParseUUIDPipe) id: string,
+    @Res() res: express.Response,
+  ): Promise<void> {
+    await this.service.getCostSurfaceCsv(id, res);
+    return;
```

Comment:
![image](https://user-images.githubusercontent.com/3466388/122350020-1c59a300-cf4d-11eb-9ee6-cbd0ef1216a3.png)

as you wrote, we don't have the HTTP context down the road.

```
actually return a stream downstream and let the final consumer deal with piping it as needed.
```
No worries, I agree that it could be kind better in terms of separation, even at a cost of controller not being so dumb any longer.

@hotzevzl Let me know if we want to put a task to refactor that tiny bit later on ❤️

Reply:
I think we can leave this as is even later.

Reply:
also `stream.PassThrough` may help to return a stream, I've done it in `cost-surface-template/db-storage.ts`

Reply:
yes, even better. simple, discreet and just what is needed for such a scenario.